src/ray/rpc/grpc_server.cc

```diff
@@ -141,11 +144,13 @@ void GrpcServer::PollEventsFromCompletionQueue(int index) {
       // SENDING_REPLY
       if (server_call->GetState() == ServerCallState::SENDING_REPLY) {
         server_call->OnReplyFailed();
+        // A new call should be suplied
+        need_new_call = true;
       }
       delete_call = true;
```

Comment:
What is the difference between `delete_call` and `need_new_call`? Maybe what we want is to put `delete_call` inside the brace above?

Reply:
the difference is need_new_call will be false when gRPC server is shut down.

Reply:
When will the gRpc server be down?

Reply:
in GrpcServer::Shutdown()

Reply:
I mean the `Shutdown()` method will never be invoked unless the gcs server is down right?
If gcs server is down there is no chance that `need_new_call` being false.